app/reducers/lendingRequests.ts

```diff
@@ -0,0 +1,52 @@
+import { createSelector } from '@reduxjs/toolkit';
+import { LendingRequest } from 'app/actions/ActionTypes';
+import { RootState } from 'app/store/createRootReducer';
+import createEntityReducer from 'app/utils/createEntityReducer';
+
+export default createEntityReducer({
+  key: 'lendingRequests',
+  types: {
+    fetch: LendingRequest.FETCH,
+    mutate: LendingRequest.CREATE,
+    delete: LendingRequest.DELETE,
+  },
+});
+
+export const selectLendingRequests = createSelector(
+  (state: RootState) => state.lendingRequests.byId,
+  (state: RootState) => state.lendingRequests.items,
+  (state: RootState) => state.lendableObjects.byId,
+  (lendingRequestsById, lendingRequestsIds, lendableObjectsById) =>
+    lendingRequestsIds.map((id) => {
+      const lendingRequest = lendingRequestsById[id];
+      const lendableObject = lendableObjectsById[lendingRequest.lendableObject];
+
+      return {
+        ...lendingRequest,
+        lendableObject,
+      };
```

Comment:
I'm not a fan of adding nested entities in selectors. Imo. the lendableObject should just be selected by the component that uses it.
I haven't looked too closely at the component code, but you probably have some list of lending requests, and then map over it to render each separate lending request. Within the lending request list object component, do a normal `selectLendingObjectById` with the id from `lendingRequest.lendableObject`. 
It makes the types easier to deal with as you don't need a new `TransformedLendingRequest` (er even two if you have separate list and detail serializers), and avoids unneccesary selecting from the store if we ever select lending requests without using the lendable objects.